proxy/src/fully_qualified_authority.rs

```diff
@@ -41,13 +41,14 @@ impl FullyQualifiedAuthority {
             }
         };
 
-        // parts should have a maximum 4 of pieces (name, namespace, svc, zone)
+        // Detect local service names in the form _name.namespace.svc.zone_. _zone_ may
+        // contain dotted labels.
```

Comment:
These comments are a lot clearer. I'd really like to see this function refactored a bit more, as I found it pretty hard to follow (especially the reliance on early returns) but that's probably out of scope for this PR.

Reply:
This is all going to get replaced very soon with stuff that is much clearer.